### DISCUSSION TITLE
AVD Configuration to support breakout cables

### DISCUSSION THREAD

**Discussion by keeganmilner** [aristanetworks/avd#4386] [Q&A] [answered]
Are the use of breakout cables supported to generate config with AVD? I have a L2LS topology and I'm looking to create a port channel from a single 100G port on each spine down to a pair of leaf switches with (4x)25G Uplinks. I have the each breakout split between the leaf switches. Example: Spine01 21/1-21/2 to Leaf01 51,52. Spine02 21/1-21/2 to Leaf01 53,54. I'm using 21/3-21/4 towards leaf02 from each spine. When I generate the config, it only creates a port channel for one interface going to Spine01 and one interface going to Spine02. The port-channel is missing the additional links. 

From an LLDP perspective, I can see all the links between the spines and leafs correctly.  

Here is an example of the leaf configuration with some info redacted:

`leaf:
  defaults:
    platform: default
    uplink_switches: [xxx-spine01, xxx-spine02]
    mlag_interfaces: [Ethernet49, Ethernet50]
    mlag_peer_ipv4_pool: 169.254.100.0/24
    spanning_tree_mode: mstp
    spanning_tree_priority: 36864

  node_groups:
    - group: leaf01_2
      mlag: True
      filter:
        tags: [MGMT]
      nodes:
        - name: xxx-leaf01
          id: 11
          mgmt_ip: 10.x.x.x/24
          uplink_switches: [xxx-spine01, xxx-spine02]
          uplink_interfaces: [Ethernet51, Ethernet52, Ethernet53, Ethernet54]
          uplink_switch_interfaces: [Ethernet21/1, Ethernet21/2, Ethernet21/1, Ethernet21/2]
          mlag_interfaces: [Ethernet49, Ethernet50]
        - name: xxx-leaf02
          id: 12
          mgmt_ip: 10.x.x.x/24
          uplink_switches: [xxx-spine01, xxx-spine02]
          uplink_interfaces: [Ethernet51, Ethernet52, Ethernet53, Ethernet54]
          uplink_switch_interfaces: [Ethernet21/3, Ethernet21/4, Ethernet21/3, Ethernet21/4]
          mlag_interfaces: [Ethernet49, Ethernet50]`

I'm assuming it isn't grouping the uplink interfaces appropriately based on having more than one per switch. I'm curious if there is a way around this or if this is a limitation in AVD. Please let me know if something isn't clear. Thanks in advance.

**Comment by ClausHolbechArista** [comment, marked as answer]
Make sure the length of `uplink_switches` match the length of `uplink_interfaces`. AVD have no way of knowing which of the interfaces are connected to which spine, so you have to put spine1 twice and spine2 twice.
Note that IP addressing can be skewed if you have a mixed number of uplinks between all your leaf switches. Make sure to set `max_uplink_switches` (unique switches) and `max_parallel_uplinks` (max uplinks going to the same spine) to the worst case. That will allocate enough IP addresses to cater for growth and flexibility within the max limits.

**Reply by keeganmilner** [reply]
Thank you for the response. After making the adjustments you recommended, all appears to look correct on the generated config. Thanks!
